File: CS/WebSite/OAuth/AccessTokenCache.cs
namespace OAuth_Provider_Basics.OAuth {
    using System;
    using DevExpress.Xpo;
    using System.Collections.Generic;
    using DevExpress.Data.Filtering;
    using DevExpress.Utils.OAuth.Provider;
    using DevExpress.Utils.OAuth;
    using System.Web;

    public class AccessTokenCache : AccessTokenStore {       
        public override IToken GetToken(string token) {
            if (String.IsNullOrEmpty(token)) {
                return null;
            }
            return this.Get("ac:" + token);
        }
        
        public override void RevokeToken(string token) {
            if (String.IsNullOrEmpty(token)) {
                return;
            }
            this.Set("ac:" + token, null);
        }

        public override IToken CreateToken(IToken requestToken) {
            if (requestToken == null || requestToken.IsEmpty) {
                throw new ArgumentException("requestToken is null or empty.", "requestToken");
            }
            Token accessToken = new Token(
                requestToken.ConsumerKey,
                requestToken.ConsumerSecret,
                Token.NewToken(TokenLength.Long),
                Token.NewToken(TokenLength.Long),
                requestToken.AuthenticationTicket,
                requestToken.Verifier,
                requestToken.Callback);
            return this.Set("ac:" + accessToken.Value, accessToken);
        }
    }
}

File: VB/WebSite/OAuth/AccessTokenCache.vb
Imports System.Web
Imports DevExpress.Utils.OAuth
Imports DevExpress.Utils.OAuth.Provider
Imports DevExpress.Data.Filtering
Imports System.Collections.Generic
Imports DevExpress.Xpo
Imports System

Namespace OAuth_Provider_Basics.OAuth

	Public Class AccessTokenCache
		Inherits AccessTokenStore

		Public Overrides Function GetToken(ByVal token As String) As IToken
			If String.IsNullOrEmpty(token) Then
				Return Nothing
			End If
			Return Me.Get("ac:" & token)
		End Function

		Public Overrides Sub RevokeToken(ByVal token As String)
			If String.IsNullOrEmpty(token) Then
				Return
			End If
			Me.Set("ac:" & token, Nothing)
		End Sub

		Public Overrides Function CreateToken(ByVal requestToken As IToken) As IToken
			If requestToken Is Nothing OrElse requestToken.IsEmpty Then
				Throw New ArgumentException("requestToken is null or empty.", "requestToken")
			End If
			Dim accessToken As New Token(requestToken.ConsumerKey, requestToken.ConsumerSecret, Token.NewToken(TokenLength.Long), Token.NewToken(TokenLength.Long), requestToken.AuthenticationTicket, requestToken.Verifier, requestToken.Callback)
			Return Me.Set("ac:" & accessToken.Value, accessToken)
		End Function
	End Class
End Namespace